src/Topics/StandardModernPrecision/Bids1C1Halt.hs

```diff
@@ -0,0 +1,243 @@
+    makeAlertableCall (T.Bid 2 T.Notrump) "1444 shape, singleton spade"
+
+
+b1C1H2C2D :: Action
+b1C1H2C2D = do
+    alternatives . map (`suitLength` 4) $ T.majorSuits
+    balancedHand
+    makeAlertableCall (T.Bid 2 T.Diamonds) "waiting bid"
+
+
+b1C1H2C2H :: Action
+b1C1H2C2H = do
+    minSuitLength T.Hearts 5
+    -- TODO: Do we need extra restrictions here on not having a club fit, or
+    -- having longer hearts than diamonds or anything?
+    makeCall (T.Bid 2 T.Hearts)
+
+
+b1C1H2C2S :: Action
+b1C1H2C2S = do
+    minSuitLength T.Diamonds 5
+    makeAlertableCall (T.Bid 2 T.Diamonds) "5+ diamonds"
+
+
+b1C1H2C2N :: Action
+b1C1H2C2N = do
+    maxSuitLength T.Clubs 3  -- TODO: is the max club length actually 2?
+    mapM_ (`maxSuitLength` 4) T.allSuits
+    balancedHand
+    forbid b1C1H2C2D
+    makeCall (T.Bid 2 T.Notrump)
+
+
+b1C1H2C3C :: Action
+b1C1H2C3C = do
+    mapM_ forbid [balancedHand, b1C1H2C2D, b1C1H2C2H]
+    minSuitLength T.Clubs 3
+    makeCall (T.Bid 3 T.Clubs)
+
+
+b1C1H2C3D :: Action
+b1C1H2C3D = do
+    suitLength T.Clubs 1
+    suitLength T.Diamonds 4
+    suitLength T.Hearts 4
+    suitLength T.Spades 4
+    makeAlertableCall (T.Bid 3 T.Diamonds) "4441 shape, singleton club"
+
+
+b1C1H2D2H :: Action
+b1C1H2D2H = do
+    minSuitLength T.Hearts 5
+    makeCall (T.Bid 2 T.Hearts)
+
+
+b1C1H2D2S :: Action
+b1C1H2D2S = do
+    alternatives . map (`suitLength` 4) $ T.majorSuits
+    balancedHand
+    makeAlertableCall (T.Bid 2 T.Spades) "waiting bid"
+
+
+b1C1H2D2N :: Action
+b1C1H2D2N = do
+    maxSuitLength T.Diamonds 3  -- TODO: is the max Diamonds length actually 2?
+    mapM_ (`maxSuitLength` 4) T.allSuits
+    balancedHand
+    forbid b1C1H2D2S
+    makeCall (T.Bid 2 T.Notrump)
+
+
+b1C1H2D3C :: Action
+b1C1H2D3C = do
+    minSuitLength T.Clubs 5
+    maxSuitLength T.Diamonds 2
+    forbid b1C1H2D2S
+    makeCall (T.Bid 3 T.Clubs)
+
+
+b1C1H2D3D :: Action
+b1C1H2D3D = do
+    mapM_ forbid [balancedHand, b1C1H2D2H, b1C1H2D2S, b1C1H2D2N]
+    minSuitLength T.Diamonds 3
+    makeCall (T.Bid 3 T.Diamonds)
+
+
+b1C1H2D3H :: Action
+b1C1H2D3H = do
+    suitLength T.Clubs 4
+    suitLength T.Diamonds 1
+    suitLength T.Hearts 4
+    suitLength T.Spades 4
+    makeAlertableCall (T.Bid 3 T.Hearts) "4414 shape, singleton diamond"
+
+
+b1C1H2H2S :: Action
+b1C1H2H2S = do
+    -- TODO: we really want to show hands where either we have 4 spades *or* we
+    -- think partner should declare in notrump. Is there an easy way to code
+    -- that latter condition?
+    suitLength T.Spades 4
+    forbid b1C1H2H3H
+    balancedHand
+    makeAlertableCall (T.Bid 2 T.Spades) "waiting bid"
+
+
+b1C1H2H2N :: Action
+b1C1H2H2N = do
+    mapM_ (`maxSuitLength` 4) T.allSuits
+    balancedHand
+    mapM_ forbid [b1C1H2H2S, b1C1H2H3H]
+    makeCall (T.Bid 2 T.Notrump)
+
+
+b1C1H2H3C :: Action
+b1C1H2H3C = do
+    maxSuitLength T.Hearts 2
+    mapM_ forbid [b1C1H2H2S, b1C1H2H2N, b1C1H2H3H]
+    minSuitLength T.Clubs 5
```

Comment:
It'll be interesting to see this in actual situations: the book predicts that the clubs are almost always 6 cards long, but I can't yet envision when it's only 5. Are you, like, 2245 or something? That sounds common, which makes me wonder if those hands should respond 2S or maybe even 2N instead of 3C.